Polygon/include/CGAL/draw_polygon_set_2.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 1997
+// Utrecht University (The Netherlands),
+// ETH Zurich (Switzerland),
+// INRIA Sophia-Antipolis (France),
+// Max-Planck-Institute Saarbruecken (Germany),
+// and Tel-Aviv University (Israel).  All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Guillaume Damiand <guillaume.damiand@liris.cnrs.fr>
+
+#ifndef CGAL_DRAW_POLYGON_SET_2_H
+#define CGAL_DRAW_POLYGON_SET_2_H
+
+#include <CGAL/draw_polygon_with_holes_2.h>
+
+#ifdef CGAL_USE_BASIC_VIEWER
+#include <CGAL/Polygon_set_2.h>
+
+namespace CGAL
+{
+
+template<class PS2>
+class SimplePolygonSet2ViewerQt :
+  public SimplePolygonWithHoles2ViewerQt<typename PS2::Polygon_with_holes_2>
+{
+  typedef SimplePolygonWithHoles2ViewerQt<typename PS2::Polygon_with_holes_2> Base;
+
+public:
+  SimplePolygonSet2ViewerQt(QWidget* parent, const PS2& aps2,
+                            const char* title="Basic Polygon_set_2 Viewer") :
+    Base(parent, title)
+  {
+    const int n = aps2.number_of_polygons_with_holes();
+    typename PS2::Polygon_with_holes_2 polygons[n];
+    aps2.polygons_with_holes(polygons);
+
+    for (typename PS2::Polygon_with_holes_2 P: polygons) {
```

Comment:
Question: do we need to use typename PS2::Polygon_with_holes_2& P?
I am not sure if typename PS2::Polygon_with_holes_2 P will copy the polygons or not.

Reply:
You're right of course; this should be a reference, as it will copy the polygn needlessly as it is.